errors.go

```diff
@@ -56,21 +58,12 @@ func (e ErrNotExists) Error() string {
 	return fmt.Sprintf("key not found: %q", e.Segment)
 }
 
-// ErrInvalidKey may be returned from lookup functions on the Node interface
-// when a key is invalid.
-//
-// Common examples of this are when `Lookup(Node)` is used with a non-string Node;
-// typed nodes also introduce other reasons a key may be invalid.
-type ErrInvalidKey struct {
-	Reason string
-
-	// Perhaps schema.ErrNoSuchField could be folded into this?
-	// Perhaps Reason could be replaced by an enum of "NoSuchField"|"NotAString"|"ConstraintRejected"?
-	// Might be hard to get rid of the freetext field entirely -- constraints may be nontrivial to describe.
+type ErrRepeatedMapKey struct {
```

Comment:
No docs, are you thinking this is self explanatory or did you just forget to add docs?